v2/tools/generator/internal/codegen/pipeline/add_secrets.go

```diff
@@ -59,6 +60,17 @@ func applyConfigSecretOverrides(config *config.Configuration, definitions astmod
 			}
 		}
 
+		// For secrets, setting this on the spec should also apply to the status
+		if typeName.IsStatus() {
```

Comment:
Rather than re-iterating, should we look into merging this into the loop above with some branches?

Reply:
Yeah, agreed that's better. Fixed.